widgetssdk/src/main/java/com/glia/widgets/view/textview/SingleLineHintEditText.kt

```diff
@@ -0,0 +1,76 @@
+package com.glia.widgets.view.textview
+
+import android.content.Context
+import android.text.Editable
+import android.text.TextUtils
+import android.text.TextWatcher
+import android.util.AttributeSet
+import androidx.appcompat.widget.AppCompatEditText
+
+/**
+ * This class is here to solve next bug:
+ * See video in this comment to understand initial bug: https://glia.atlassian.net/browse/MOB-3860?focusedCommentId=459636
+ *
+ * The issue was caused by default EditText behavior. The hint height is effecting the EditText height even when the hint is not displayed.
+ * This means that if for some reason your EditText view has property maxLines = 4 and your hint is so long that it takes 4 lines and
+ * then when user write just 1 letter into your EditText it would still occupy as much height as if it had 4 lines of code.
+ *
+ * Another related problem is that for some reason at least in ConstraintLayout multi-line hint messes up layout rules. See next comment
+ * in the same ticket: https://glia.atlassian.net/browse/MOB-3860?focusedCommentId=459642.
+ * To prevent this from happening this implementation forces hint to always be one-line
+ */
+internal class SingleLineHintEditText @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : AppCompatEditText(context, attrs, defStyleAttr) {
+
+    private var hintOverride: CharSequence? = null
+    var maxLinesOverride: Int = 0
+    var ellipsizeOverride: TextUtils.TruncateAt? = null
+
+    init {
+        hintOverride = hint           // Retrieves existing value if it was set in XML
+        maxLinesOverride = maxLines   // Retrieves existing value if it was set in XML
+        ellipsizeOverride = ellipsize // Retrieves existing value if it was set in XML
+
+        if (shouldShowHint()) {
+            showHintAndMakeItOneLine()
+        }
+        addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+                /* no-op */
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                /* no-op */
+            }
+
+            override fun afterTextChanged(s: Editable?) {
+                if (shouldShowHint()) {
+                    showHintAndMakeItOneLine()
+                } else if (!s.isNullOrEmpty() && !hint.isNullOrEmpty()) {
+                    removeHintAndRestoreDefaultSettings()
+                }
+            }
+
+        })
+    }
+
+    fun setHintOverride(s: CharSequence?) {
+        hintOverride = s
+        if (shouldShowHint()) {
+            showHintAndMakeItOneLine()
+        }
+    }
+
+    private fun shouldShowHint(): Boolean = text.isNullOrEmpty() && hint != hintOverride
+
+    private fun removeHintAndRestoreDefaultSettings() {
+        hint = ""
+        maxLines = maxLinesOverride
+        ellipsize = ellipsizeOverride
+    }
```

Comment:
It seems to me that this function name is confusing - what is meant by 'default': Android `EditText` settings or those that are overridden? At first I thought about Android settings. Hm, maybe we can just drop the word `default`?
`removeHintAndRestoreDefaultSettings` => `removeHintAndRestoreSettings` or `removeHintAndRestoreEditTextSettings`

Up to you. I just shared how I interpreted it initially.

Reply:
I agree with what you said. 'Default' could cause a lot of confusion here.
Will use  `removeHintAndRestoreEditTextSettings `

Reply:
Fixed in 7dc2b11